plugins/orchestrator/src/components/WorkflownstancesViewerPage/ProcessDetailsViewer.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useMemo } from 'react';
+import Moment from 'react-moment';
+
+import { InfoCard } from '@backstage/core-components';
+
+import { Button, Link, Typography } from '@material-ui/core';
+import CancelOutlinedIcon from '@material-ui/icons/CancelOutlined';
+import CheckCircleIcon from '@material-ui/icons/CheckCircle';
+import GetAppIcon from '@material-ui/icons/GetApp';
+import LaunchIcon from '@material-ui/icons/Launch';
+import PauseCircleFilledRoundedIcon from '@material-ui/icons/PauseCircleFilledRounded';
+import PlayCircleFilledWhiteRoundedIcon from '@material-ui/icons/PlayCircleFilledWhiteRounded';
+import PublishIcon from '@material-ui/icons/Publish';
+import RemoveCircleRoundedIcon from '@material-ui/icons/RemoveCircleRounded';
+
+import {
+  ProcessInstance,
+  ProcessInstanceState,
+} from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { Paragraph } from '../Paragraph/Paragraph';
+import ItemDescriptor from './ItemDescriptor';
+
+interface ProcessDetailsViewerProps {
+  selectedInstance: ProcessInstance | undefined;
+}
+
+const processInstanceIconCreator = (state: string) => {
+  const render = (icon: JSX.Element, text: string) => {
+    return (
+      <Paragraph>
+        <div
+          style={{
+            display: 'flex',
+            alignItems: 'center',
+            flexWrap: 'wrap',
+          }}
+        >
+          {icon}
+          <Typography
+            component="span"
+            variant="body2"
+            style={{ paddingLeft: '8px' }}
+          >
+            {text}
+          </Typography>
+        </div>
+      </Paragraph>
+    );
+  };
+
+  switch (state) {
+    case ProcessInstanceState.Active:
+      return render(<PlayCircleFilledWhiteRoundedIcon />, 'Active');
+    case ProcessInstanceState.Completed:
+      return render(<CheckCircleIcon htmlColor="#3e8635" />, 'Complete');
+    case ProcessInstanceState.Aborted:
+      return render(<RemoveCircleRoundedIcon />, 'Aborted');
+    case ProcessInstanceState.Suspended:
+      return render(<PauseCircleFilledRoundedIcon />, 'Suspended');
+    case ProcessInstanceState.Error:
+      return render(<CancelOutlinedIcon color="error" />, 'Error');
+    default:
+      return <></>;
+  }
+};
+
+const getProcessInstanceDescription = (processInstance: any) => {
+  return {
+    id: processInstance.id,
+    name: processInstance.processName,
+    description: processInstance.businessKey,
+  };
+};
+
+export const ProcessDetailsViewer = (props: ProcessDetailsViewerProps) => {
+  const { selectedInstance } = props;
+
+  const errorInfo = useMemo(() => {
```

Comment:
We should avoid memoizing React elements.
It would be better to memoize prepared data and then use them to conditionally create a tree of React elements.

Reply:
There is a refactoring going on and this component will likely be replaced with new code.